### PR TITLE
Update the README after the cache removal

### PR DESCRIPTION
**Pull request by jwbixby**
Removing these mentions of the `cache_result` feature after its removal in version `1.0.0`.

### CODE REVIEW

**Comment by simon-weber** [comment]
Looks like there's also a typo in setup.py: `s/History/HISTORY`: https://travis-ci.org/venmo/business-rules/jobs/116436572.

No clue why it didn't fail locally when I packaged it.

**Comment by jwbixby** [comment]
So weird. The tests passed too on that other PR

**Comment by simon-weber** [comment]
(whoops -- we were running on osx's case-insensitive filesystem when packaging and testing the install)

File: README.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ class ProductVariables(BaseVariables):
         last_order = self.product.orders[-1]
         return (last_order.expiration_date - datetime.date.today()).days
 
-    @string_rule_variable(cache_result=False)
+    @string_rule_variable
```
Comment (simon-weber) [inline README.md]: I think the parens are still necessary for this decorator (it's not compatible with both styles of calling).

**Comment by simon-weber** [comment]
:green_apple:

**Pull request merged**